### DISCUSSION TITLE
Possibility to use subfolder in URL

### DISCUSSION THREAD

**Discussion by madcoda9000** [photoview/photoview#518] [Q&A and Troubleshooting]
Hello,

me again :-)

I've installed photoviw manually using your guide. My Photoview installation is reachable now at http://localhost:4001.

Would it be possible to let it run on address http://localhost:4001/photoview ??

Greetings
Sascha

**Comment by viktorstrate** [comment]
See #507

You can't do that without a reverse-proxy though.

**Reply by madcoda9000** [reply]
hehe, i tryed it with a reverse proxy! but then the page stays empty (without an error message).

I use the following rewrite rule:

<Location /photoview>
                        RewriteEngine On
                         RewriteCond %{HTTP:Upgrade} =websocket [NC]
                         RewriteRule /notifications/hub(.*) ws://localhost:3012/$1 [P,L]
                         ProxyPass http://localhost:4001/
                         ProxyPreserveHost On
                         RequestHeader set X-Real-IP %{REMOTE_ADDR}s
</Location>

**Reply by viktorstrate** [reply]
I have little experience with Apache, try to ask on the discord channel

**Reply by madcoda9000** [reply]
ok i try. many thx

**Comment by madcoda9000** [comment]
So, I've tested a lot. It seems to be an issue with the application. If i open the application with a reverse proxy over the web root (for ex. internal port 4001, external port 443. The ist is working.

But if i try to use a subfolder, i only het a blank page. If i try to use manually mydomain/photoview/photos, it reverts automatically to mydomain/photos. So it seems that the webapp is not usable for such an scenario :-(

Now i'll try to open the app on an ohter https port. Maybe that will work.

**Comment by madcoda9000** [comment]
Ok, now i did the following to get it working beside my other appliction running on port 443 already.

1. do a manually install
2. create a new ssl enabled virtual web on an other port (i've used 4002). This is the apache config i am using:

```
 1 <IfModule mod_ssl.c>
  2         <VirtualHost photoview:4002>
  3                 ServerAdmin webmaster@localhost
  4
  5                 DocumentRoot /var/www/html
  6
  7                 <Location />
  8                         RewriteEngine On
  9                         RewriteCond %{HTTP:Upgrade} =websocket [NC]
 10                         RewriteRule /notifications/hub(.*) ws://localhost:3012/$1 [P,L]
 11                         ProxyPass http://localhost:4001/
 12                         ProxyPreserveHost Off
 13                         RequestHeader set X-Real-IP %{REMOTE_ADDR}s
 14                 </Location>
 15
 16                 ProxyPassReverse / http://localhost:4001
```


3. open the port in the firewall
4. now photoview ie reachable by the url: https://yourdomain:4002

Ok, not the best solution, but it is working. Maybe you can modify your app, so that it can be run as a subfolder in the future :-) 

Greetings 
Sascha
